lib/raw-stream-set.js

```diff
@@ -18,12 +20,35 @@ module.exports = class RawStreamSet {
     const stream = this._dht._udx.createStream(id, opts)
     this._streams.set(id, stream)
 
-    stream.on('close', onclose)
+    stream
+      .on('close', onclose)
+      .on('connect', onconnect)
 
     return stream
 
     function onclose () {
       self._streams.delete(id)
     }
+
+    async function onconnect () {
+      const {
+        remoteHost: host,
+        remotePort: port
+      } = stream
+
+      for (let mtu = 1400; mtu > 1200; mtu -= 100) {
+        try {
+          await self._dht.ping({ host, port }, {
+            size: mtu - 12, /* overhead */
+            socket: stream.socket
+          })
+
+          stream.setMTU(mtu)
+          break
+        } catch (err) {
+          safetyCatch(err)
```

Comment:
@kasperisager can you test what happens if this runs when the socket has been closed?

Reply:
ie, if the socket is opened and closed and then we run this ping in the background, just to make sure we don't crash becaue of that

Reply:
I'm not sure if I can construct a case where the socket is closed before the ping is attempted 🤔 The `catch` clause is hit during the other test cases though.

Reply:
@mafintosh Let me know if you have pointers on the above ☝️